### DISCUSSION TITLE
Can a near-cache also expose its back-tier using Extend/gRPC?

### DISCUSSION THREAD

**Discussion by javafanboy** [oracle/coherence#125] [Q&A] [answered]
I am using a near-cache to let container based services share data efficiently but now I would like to allow also Java based Lambda functions access the same data - before I try to figure out how to configure this (figuring out what the XML config should look like is always my main stumbling block when using Coherence) I would like to know if this even is possible or do Extend only work with remote caches not part of a "near" setup?

**Comment by javafanboy** [comment]
I rephrased this as an "issue" as I have had more luck getting those answered :-)

**Reply by lsho** [reply]
Yes, extend client can have near cache in which case, the ```<back-scheme>``` would be a ```<remote-cache-scheme>```.  Here is an example (my-remote is a ```<remote-cache-scheme>```):

```
    <near-scheme>
      <scheme-name>my-near-present</scheme-name>
      <front-scheme>
        <local-scheme>
          <high-units>1000</high-units>
        </local-scheme>
      </front-scheme>
      <back-scheme>
        <remote-cache-scheme>
          <scheme-ref>my-remote</scheme-ref>
        </remote-cache-scheme>
      </back-scheme>
      <invalidation-strategy>present</invalidation-strategy>
    </near-scheme>
```

**Reply by javafanboy** [reply]
Thanks but what you posted above is very similar to what I already have in my cache config for my normal near cache but I do not see how it related to accessing the SAME cache also using gRPC?!
I posted this question also under issues as it seems more people are reading/responding there than here under discussions. I posted some program examples (and my cache config) to better explain what I am trying to achive.
Can you please have a look there and if you have some suggestions post as a reply to that issue so we get all discussions in one place....

**Comment by ghost** [comment, marked as answer]
Hello @javafanboy I'm guessing this is related to this issue: https://github.com/oracle/coherence/issues/126.  Is that correct?  If you're all set, would you mind marking this as answered?
